Assets/Code/Scripts/NoclipRealityManagement/RealityMovementCalibration.cs

```diff
@@ -0,0 +1,489 @@
+    private float _horizontalInput;
+    private float _verticalInput;
+
+    private Vector3 _moveDirection;
+
+    private Transform _transform;
+    private Rigidbody _rigidbody;      // set the rigidbody
+    private NoclipManager _noclipManager;
+
+    // It's true if is the realbody, it's false if is the noclip body
+    private bool _currentPlayerBody  = true;
+
+    // player states
+    [SerializeField] private MovementState _state;     // current player state
+
+    [SerializeField] private bool _OnSlope; //This seriazlized field is only use to debug from unity // TODO remove
+    [SerializeField] private bool Grounded; //This seriazlized field is only use to debug from unity // TODO remove
+    
+
+    //original gravity
+    private Vector3 _originalGravity;
+    private void Awake()
+    {
+        _rigidbody = GetComponent<Rigidbody>();
+        _transform = GetComponent<Transform>();
+        _noclipManager = FindObjectOfType<NoclipManager>();
+    }
+
+    // Start is called before the first frame update
+    private void Start()
+    {
+
+        _rigidbody.freezeRotation = true;
+        
+        ResetJump(); // set _readyToJump to "true"
+
+        _startYScale = _transform.localScale.y;
+
+        //if gravity magnitude is not 0 (Gamemanager zeroes it on game boot)
+        if (Physics.gravity.magnitude != 0)
+        {
+            //set gravity to gravity magnitude
+            Physics.gravity = new Vector3(0, -_gravity, 0) * _gravityMultiplier;
+        }
+    }
+
+    // Update is called once per frame
+    private void Update()
+    {
+        //IMPORTANT LESSON materials loaded at runtime should be instantiated
+        //find ErrorCube and enable its mesh
+        //GameObject errorCube = GameObject.Find("ErrorCube");
+        //find realitybody
+        //GameObject realityBody = GameObject.Find("RealityBody");
+        //set material of errorCube to material of RealityBody
+        //errorCube.GetComponent<MeshRenderer>().material = realityBody.GetComponent<MeshRenderer>().material;
+
+        //if (_calibration) 
+        CalibrationMenu();
+
+        if (!_noclipManager.IsNoclipEnabled())
+        {
+            //Alternative 1
+            //_grounded = Physics.Raycast(_transform.position,  Vector3.down, _playerHeight * 0.5f + 0.3f, _ground);
+        
+            //Alternative 2
+            // ATTENTION!!! the 0.48f value is based on the player capsule radius
+            _grounded = Physics.CheckSphere(_groundCheck.position,  0.48f, _ground);
+        
+            // stefanofossati comment: I Think that this second alternative is a little more precise for the slopes
+            _OnSlope = OnSlope(); // used to debug // TODO remove
+            Grounded = _grounded; // used to debug // TODO remove
+        
+            UserInput();
+            StateHandler();
+            MovePlayer();
+            SpeedControl();
+
+            // handle drag
+            if (_grounded && !_commitJump)
```

Comment:
Yesterday we tried to add this part in order to have the drag only when any keys aren't pressed.
@AleDeNic said that is better in this way.

```suggestion
            if (_grounded && && _horizontalInput == 0 && _verticalInput == 0 &&!_commitJump)
```